[Feature] User 프로필 UI 구현

dudtjs1021ej opened the pull request:
### 📙 작업 내역

- [x] User용 프로필 UI 구현
- [x] User용 게시물 UI 구현
- [x] 계정신고 UI 구현
- [x] 커스텀 backButton 사용을 위해 UINavigationController + extension 추가
- [x] 기존 ProfileViewController -> MyProfileViewController 네이밍 변경 


### 📘 작업 유형

- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트


### 📋 체크리스트

- [ ] Merge 하는 브랜치가 올바른가?
- [ ] PR과 관련없는 변경사항이 없는가?
- [ ] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

### 📱 테스트 방법

- 메인화면의 코디셀 터치 시 게시물로 이동합니다.
- 메인화면의 코디셀의 프로필뷰 터치 시 유저 프로필로 이동합니다.

### 📝 리뷰 노트

- 코디네이터패턴 이상한 부분 바로 말해주세요...!!
- 계정신고버튼을 누르고 dismiss되는게 좀 어색한데 animated를 false로 둘지..? 아니면 자연스럽게 dismiss하는 방법이 있는지 고민입니다🥲
- backButton 커스텀하기 위해 UINavigationController + extension 추가했습니다!

### 🏞️ 스크린샷


![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/77915491/213207864-89d28d47-5848-4e6b-9b93-cdfeb06dffbd.gif)

dudtjs1021ej committed:
✨ [feat] UserCoordinator 구현

dudtjs1021ej committed:
✨ [feat] UserViewController 네비게이션바 구현

dudtjs1021ej committed:
✨ [feat] UserProfile 구현

- Profile -> MyProfile 네이밍 변경
- User -> UserProfile 네이밍 변경

dudtjs1021ej committed:
✨ [feat] UINavigationController+extension 추가

- customBackButton 구현

dudtjs1021ej committed:
✨ [feat] User 게시물 구현

dudtjs1021ej committed:
✨ [feat] 메인화면의 코디 터치 시 게시물 이동 구현

dudtjs1021ej committed:
✨ [feat] 계정 신고창 구현

dudtjs1021ej committed:
🐛 [fix] ProfileInfoView 레이아웃 충돌 해결

Daltonicc commented on Fitfty/Projects/Coordinator/Sources/Main/MainCoordinator.swift:
```diff
@@ -50,6 +50,13 @@ private extension MainCoordinator {
+        let coordinator = UserCoordinator(navigationConrtoller: navigationController)
+        coordinator.parentCoordinator = self
+        coordinator.finishDelegate = self
+        return coordinator
```
[P3] childCoordinators에 userCoordinator가 append 안되어 있는 것 같아요~

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/14/commits/6dd7e64c95267e5bb3ef97a112b8770a3e2e6c67
수정했습니다😁

Daltonicc commented on Fitfty/Projects/Profile/Sources/MyProfile/Views/Profile/ProfileView.swift:
```diff
@@ -0,0 +1,72 @@
+    
+    private lazy var contentLabel: UILabel = {
+        let label = UILabel()
+        label.text = "안녕하세요! 반갑습니다"
```
[P3 - 하지만 추후에 챙겨도 무방!] 지금은 임시로 하드코딩해두신거겠지만 nicknameLabel이나 contentLabel 모두 유저에 의해 텍스트가 바뀔 수 있으니까 setNicknameLabelText(_ text: String) 이런 식으로 미리 메서드로 빼놓는게 좋을 거 같아요 ! 뷰모델 작업할 때 챙기셔도 무방합니당

dudtjs1021ej replied:
setUp 메소드는 작성해두었는데 이 부분을 실수로 안지웠네여..!! 수정하겠습니다!

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/14/commits/10a7e81d5ad5976a820cbe1ef3ca84b7ff819708
제거 했습니당👍

Daltonicc commented on Fitfty/Projects/Profile/Sources/MyProfile/Views/Profile/ProfileView.swift:
```diff
@@ -0,0 +1,72 @@
+        let label = UILabel()
+        label.text = "안녕하세요! 반갑습니다"
+        label.font = FitftyFont.appleSDMedium(size: 15).font
+        label.textColor = UIColor(red: 0.512, green: 0.512, blue: 0.567, alpha: 1)
```
[질문] 썬~~ 혹시 여기는 왜 rgb값을 직접 넣어주신걸까요?? 컬러셋에 없는 색깔일까요??

dudtjs1021ej replied:
예전에 구현 해놓은 프로필 관련 뷰는 아직 공통 컬러셋으로 수정하지 못했어요🥲 (ProfileView의 파일 위치를 수정했더니 이번 PR에 포함되었네여..)
지금 작업하고 있는 프로필 UI 수정에서 한번에 진행할 예정입니다!

Daltonicc commented on Fitfty/Projects/Coordinator/Sources/Profile/UserCoordinator.swift:
```diff
@@ -0,0 +1,73 @@
+    func showReport(_ viewController: UIViewController) {
+        let reportViewController = makeReportViewController()
+        reportViewController.modalPresentationStyle = .overFullScreen
+        viewController.present(reportViewController, animated: false)
```
[질문] 썬~ 혹시 요거 viewController를 파라미터로 받고 있는 이유가 있을까요? navigationController.present(~~) 로 하지 않고 직접 받으신 이유가 궁금합니당

dudtjs1021ej replied:
```swift
navigationController.present(reportViewController, animated: true)
```
정확한 이유는 모르겠지만 이렇게 작성하면 present에 약간의 딜레이가 생기더라구여..
그래서 파라미터로  viewController를 받아와 present 해주었습니다!

Daltonicc commented on Fitfty/Projects/Coordinator/Sources/Profile/UserCoordinator.swift:
```diff
@@ -0,0 +1,73 @@
+    }
+    
+    func dismissReport(_ viewController: UIViewController) {
+        viewController.dismiss(animated: true)
```
[질문] 여기도 위 질문과 동일합니다 !

dudtjs1021ej replied:
위와 같은 이유로 똑같이 viewController를 받아왔습니다!

Daltonicc commented on Fitfty/Projects/Profile/Sources/UserProfile/ViewControllers/UserPostViewController.swift:
```diff
@@ -0,0 +1,56 @@
+    
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setNavigationBar()
```
[질문] 썬~ 네비바 세팅만 viewWillAppear에서 해주신 이유가 궁금합니다~~

dudtjs1021ej replied:
viewDidLoad가 호출될 땐 navigationController가 nil이라 viewWillAppear에서 셋팅 해주었습니다!

Daltonicc reviewed (commented):
썬 고생하셨습니다!!!😄 질문이랑 코멘트 몇 개있는데 확인 부탁드려요 !!

계정신고뷰가 dismiss될 때 딱히 저는 어색하다는 느낌은 못받았어요 ! 어떤 부분이 걸리시는 걸까요?

dudtjs1021ej committed:
🐛 [fix] childCoordinators append 코드 추가

dudtjs1021ej committed:
♻️ [refactor] 불필요한 label text 초기화 코드 제거

dudtjs1021ej reviewed (commented):
@Daltonicc 에단 리뷰 감사합니다👍 
전 계정신고 버튼을 누르면 dismiss 될때 뒤에 검정 배경이 따라가는 부분..?이 조금 어색하다 느껴졌는데 나중에 만나서 얘기해봐도 될 것 같아여!
질문에 답글 달아두었고 추가적으로 개선해야할 부분이 있으면 바로 반영하겠습니다😁

dudtjs1021ej committed:
Merge branch 'develop' of https://github.com/YAPP-Github/21st-iOS-Team-2-iOS into feature/User-ios2-129

dudtjs1021ej committed:
🐛 [fix] merge 이후 에러 해결

- TabCoordinator의 불필요한 BaseNavigationController 코드 제거
- MainViewController의 didSelectItemAt메소드 중복 제거
- UserCoordinator의 navigationController를 BaseNavigationController로 수정

Daltonicc reviewed (approved):
넵 고생하셧습니다 ! 머지가시져
